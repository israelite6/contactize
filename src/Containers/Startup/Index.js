import React, {useCallback, useState} from 'react';
import {
  Button,
  PermissionsAndroid,
  View,
  TextInput,
  Modal,
  FlatList,
} from 'react-native';
import Contacts from 'react-native-contacts';
import ContactListItem from '../../Components/ContactListItem';

export default function Startup(props) {
  const [state, setState] = useState({
    isModalVisible: false,
    contacts: [],
    phoneNumber: '',
  });

  const handleSetState = useCallback((data) => {
    setState((prev) => ({...prev, ...data}));
  }, []);

  const handleOpenContact = useCallback(() => {
    if (state.contacts.length === 0) {
      androidPermission();
    } else {
      handleSetState({isModalVisible: true});
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state]);

  const androidPermission = async () => {
    try {
      const granted = await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.READ_CONTACTS,
      );
      if (granted === PermissionsAndroid.RESULTS.GRANTED) {
        handleGetContacts();
      }
    } catch (err) {}
  };

  const handleGetContacts = useCallback(() => {
    Contacts.getAll()
      .then((contacts) => {
        console.log(contacts);
        handleSetState({isModalVisible: true, contacts});
      })
      .catch((e) => {
        console.log(e);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleSelectPhoneNumber = useCallback((phoneNumber) => {
    handleSetState({phoneNumber, isModalVisible: false});
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const renderItem = useCallback(({item}, index) => {
    return <ContactListItem onPress={handleSelectPhoneNumber} {...item} />;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <View>
      <Button title="Open Contact" onPress={handleOpenContact} />
      <TextInput value={state.phoneNumber} />
      <Modal visible={state.isModalVisible}>
        <FlatList
          keyExtractor={(item) => item.rawContactId}
          renderItem={renderItem}
          data={state.contacts}
        />
      </Modal>
    </View>
  );
}
