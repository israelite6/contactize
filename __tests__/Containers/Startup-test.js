/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Startup from '../../src/Containers/Startup/Index';

it('Startup container renders correctly', () => {
  const tree = renderer.create(<Startup />).toJSON();
  expect(tree).toMatchSnapshot();
});
