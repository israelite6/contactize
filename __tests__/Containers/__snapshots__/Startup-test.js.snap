// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Startup container renders correctly 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open Contact
      </Text>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />
  <Modal
    hardwareAccelerated={false}
    visible={false}
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </Modal>
</View>
`;
